frontend/src/components/common/FilterBox.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component, PropTypes } from 'react'
+
+import onClickOutsideAddon from 'react-onclickoutside'
+
+class FilterBox extends Component {
+  static propTypes = {
+    topic: PropTypes.object,
+    map: PropTypes.object,
+    filterData: PropTypes.object,
+    allForFiltering: PropTypes.object,
+    visibleForFiltering: PropTypes.object,
+    toggleMetacode: PropTypes.func,
+    toggleMapper: PropTypes.func,
+    toggleSynapse: PropTypes.func,
+    filterAllMetacodes: PropTypes.func,
+    filterAllMappers: PropTypes.func,
+    filterAllSynapses: PropTypes.func,
+    closeBox: PropTypes.func
+  }
+
+  handleClickOutside = () => {
+    this.props.closeBox()
+  }
+
+  render () {
+    const { topic, map, filterData, allForFiltering, visibleForFiltering, toggleMetacode,
+            toggleMapper, toggleSynapse, filterAllMetacodes,
+            filterAllMappers, filterAllSynapses } = this.props
+    const style = {
+      maxHeight: document.body.clientHeight - 108 + 'px'
```

Comment:
this threw me off a bit. i'm being nitpicky, but what about

```
maxHeight: `${document.body.clientHeight - 100}px`
```

Actually, looking at that, it's not any nicer. What about

```
const maxHeightInPixels = document.body.clientHeight - 100
const style = { maxHeight: `${maxHeightInPixels}px` }
```

man. there is no good way to do this. I'm not sure of anything anymore